scripts/bulk-index.js

```diff
@@ -252,62 +260,74 @@ const printProgress = (count, total, startTime) => {
  *  Reindex a bunch of bibs based on a BibService query
  */
 const updateBibs = async () => {
-  await initPools()
-
   let sqlFromAndWhere = null
   let sqlParams = []
-  if (argv.bibId) {
-    sqlFromAndWhere = `bib B
-      WHERE B.nypl_source = $1
-      AND B.id = $2`
-    sqlParams = [
-      argv.nyplSource,
-      argv.bibId
-    ]
-  } else if (argv.type === 'bib' && argv.hasMarc) {
-    sqlFromAndWhere = `bib B,
-      json_array_elements(B.var_fields::json) jV
-      WHERE B.nypl_source = $1
-      AND jV->>'marcTag' = $2`
-    sqlParams = [
-      argv.nyplSource,
-      argv.hasMarc
-    ]
-  }
 
-  const query = `SELECT * FROM ${sqlFromAndWhere}` +
-    (argv.orderBy ? ` ORDER BY ${argv.orderBy}` : '') +
-    (argv.limit ? ` LIMIT ${argv.limit}` : '') +
-    (argv.offset ? ` OFFSET ${argv.offset}` : '')
+  let cursor
+  let bibClient
+
+  await newrelic.startBackgroundTransaction('Bulk-index initial query', async () => {
+    await initPools()
+
+    if (argv.bibId) {
+      sqlFromAndWhere = `bib B
+        WHERE B.nypl_source = $1
+        AND B.id = $2`
+      sqlParams = [
+        argv.nyplSource,
+        argv.bibId
+      ]
+
+      argv.limit = 1
+    } else if (argv.type === 'bib' && argv.hasMarc) {
+      sqlFromAndWhere = `bib B,
+        json_array_elements(B.var_fields::json) jV
+        WHERE B.nypl_source = $1
+        AND jV->>'marcTag' = $2`
+      sqlParams = [
+        argv.nyplSource,
+        argv.hasMarc
+      ]
+    }
+
+    const query = `SELECT * FROM ${sqlFromAndWhere}` +
+      (argv.orderBy ? ` ORDER BY ${argv.orderBy}` : '') +
+      (argv.limit ? ` LIMIT ${argv.limit}` : '') +
+      (argv.offset ? ` OFFSET ${argv.offset}` : '')
 
-  console.log(`Querying BibService: ${query} | ${JSON.stringify(sqlParams)}`)
+    console.log(`Querying BibService: ${query} | ${JSON.stringify(sqlParams)}`)
 
-  const bibClient = await dbConnectionPools.bibService.connect()
-  const cursor = bibClient.query(new Cursor(query, sqlParams))
+    bibClient = await dbConnectionPools.bibService.connect()
+    cursor = bibClient.query(new Cursor(query, sqlParams))
+  })
 
   const total = argv.limit
   let count = 0
   const startTime = new Date()
   while (count < argv.limit || !argv.limit) {
-    // Log out progress os far:
-    printProgress(count, total, startTime)
+    await newrelic.startBackgroundTransaction('Bulk-index batch', async () => {
+      // Log out progress os far:
+      printProgress(count, total, startTime)
```

Comment:
So if there was 1000 bibs getting reindexed, would this print 1000 times? Maybe we want to have a max number of prints, like 100 or so, to keep logs legible. eg, for bib count of 100 , printProgress might happen for every 10 bibs, for 1000, every 100.

Reply:
This call is made [on each batch](https://github.com/NYPL/research-catalog-indexer/blob/95f28a4f00cbc88a29e89a12758b641dafc88612/scripts/bulk-index.js#L313) so is fired every `argv.batchSize` records ([default 100](https://github.com/NYPL/research-catalog-indexer/blob/95f28a4f00cbc88a29e89a12758b641dafc88612/scripts/bulk-index.js#L46))